File: project/core/module/rest-service/src/main/java/vdi/service/rest/model/UserDetails.java
package vdi.service.rest.model;

import org.gusdb.oauth2.client.veupathdb.UserInfo;
import vdi.model.meta.UserID;

public record UserDetails(
  UserID userID,
  String firstName,
  String lastName,
  String email,
  String organization
) {
  public UserDetails(UserInfo info) {
    this(
      UserID.newUserID(info.getUserId()),
      info.getFirstName(),
      info.getLastName(),
      info.getEmail(),
      info.getOrganization()
    );
  }
}

File: service/rest-service/src/main/kotlin/org/veupathdb/service/vdi/model/UserDetails.kt
package org.veupathdb.service.vdi.model

import org.veupathdb.vdi.lib.common.field.UserID

data class UserDetails(
  val userID: UserID,
  val firstName: String?,
  val lastName: String?,
  val email: String?,
  val organization: String?,
)